core/queryparser/sparql/src/main/java/org/eclipse/rdf4j/query/parser/sparql/PrefixDeclProcessor.java

```diff
@@ -71,8 +86,18 @@ public static Map<String, String> process(ASTOperationContainer qc) throws Malfo
 			prefixMap.put(prefix, iri);
 		}
 
+		int preDefaultPrefixes = 0;
+
+		// insert the default prefixes if presents
+		if (customDefaultPrefixes != null && !customDefaultPrefixes.isEmpty()) {
+			for (Entry<String, String> defaultPrefix : customDefaultPrefixes.entrySet()) {
+				preDefaultPrefixes += insertDefaultPrefix(prefixMap, defaultPrefix.getKey(), defaultPrefix.getValue());
+			}
+		}
+
 		// insert some default prefixes (if not explicitly defined in the query)
-		final int defaultPrefixesAdded = insertDefaultPrefix(prefixMap, "rdf", RDF.NAMESPACE)
+		final int defaultPrefixesAdded = preDefaultPrefixes
```

Comment:
A non-null map at this point could replace `preDefaultPrefixes` with `customDefaultPrefixes.size()` rather than requiring a separate integer to track the number of custom prefixes.

Reply:
the `insertDefaultPrefix(...)` method is returning 1 only if the prefix was added in the `prefixMap`, so we can't use `customDefaultPrefixes.size()` because maybe a custom default prefix was defined by the user in the query.